parameters.cpp

```diff
@@ -144,216 +145,278 @@ uint P::amrMaxVelocityRefLevel = 0;
 Realf P::amrRefineLimit = 1.0;
 Realf P::amrCoarsenLimit = 0.5;
 string P::amrVelRefCriterion = string("");
-int P::amrMaxSpatialRefLevel = 0;
-int P::amrBoxHalfWidthX = 1;
-int P::amrBoxHalfWidthY = 1;
-int P::amrBoxHalfWidthZ = 1;
+uint P::amrMaxSpatialRefLevel = 0;
+uint P::amrBoxHalfWidthX = 1;
+uint P::amrBoxHalfWidthY = 1;
+uint P::amrBoxHalfWidthZ = 1;
```

Comment:
Let's hope this doesn't introduce int vs uint warnings.

Reply:
I actually believe it fixes them, instead of introducing them.

Reply:
There are several places where we have loops over AMR widths and levels with local variable of type int. I thought `auto` would do the work, but in fact I had to declare them as `uint` to avoid the warnings. Some of them have already been modified in this commit. I will check again.

Reply:
Perfect then. Anyway this is used rarely and I don't think it will cause issues as this won't overflow, nor will it be set to negative values.